OpenSim/Actuators/VandenBogert2011Muscle.cpp

```diff
@@ -0,0 +1,1234 @@
+    //double activ = getStateVariableValues(s,"activation");  // does not work
+    //double fiberLength = getStateVariableValues(s,"fiber_length");
+
+    SimTK::Vector stateVec =getStateVariableValues(s);
+    double fiberLength =stateVec[0];
+    double activ =stateVec[1];
+
+
+    double projFibLenNorm =
+            fiberLengthToProjectedLength(fiberLength/getOptimalFiberLength(),1);
+
+    VandenBogert2011Muscle::ImplicitResidual Results= calcImplicitResidual(
+            muscleLength, projFibLenNorm, activ, projFibVelNorm_guess,
+            activdot_guess, u, 0);
+
+    return Results;
+}
+
+
+//------------------------------------------------------------------------------
+VandenBogert2011Muscle::ImplicitResidual
+VandenBogert2011Muscle::calcImplicitResidual(SimTK::Vec2 y,
+                                             SimTK::Vec2 ydot_guess,
+                                             double muscleLength,
+                                             double u, int returnJacobians=0)
+const
+{
+    // Overload method for state vectors as parameters
+    VandenBogert2011Muscle::ImplicitResidual Results= calcImplicitResidual(
+            muscleLength, y[0], y[1], ydot_guess[0], ydot_guess[1], u,
+            returnJacobians);
+    return Results;
+
+}
+
+//------------------------------------------------------------------------------
+VandenBogert2011Muscle::ImplicitResidual VandenBogert2011Muscle::
+calcImplicitResidual(double muscleLength, double projFibLenNorm, double activ,
+                     double projFibVelNorm, double activdot, double u,
+                     int returnJacobians=0) const {
+
+
+    //TODO: Match symbols to doxygen diagram and Add symbolic equations comments
+    //TODO: May want to make this into a seperate function
+
+    // -------------------------Parameters----------------------------//
+    //F_{iso} Maximum isometric force that the fibers can generate
+    double maxIsoForce = getMaxIsometricForce();
+
+    //u_{max} (dimensionless) strain in the series elastic element at load of
+    //      maxIsometricForce
+    double fMaxTendonStrain = getFMaxTendonStrain(); //Strain in the series
+    //      elastic element at load of Fmax
+
+    //W (dimensionless) width parameter of the force-length relationship of
+    //      the muscle fiber
+    double fl_width = getFlWidth();
+
+    //AHill (dimensionless) Hill parameter of the force-velocity relationship
+    double fv_AHill= getFvAHill();   //Hill parameter of the f-v relationship
+
+    //FV_{max} (dimensionless) maximal eccentric force
+    double fv_maxMultiplier = getFvmaxMultiplier();
+    //Maximal eccentric force multiplier
+
+    // L_{opt}  (m) Optimal Length of Contractile Element;
+    double optFiberLength = getOptimalFiberLength();
+    //Optimal Length of Contractile Element
+
+    //b (s/m) damping coefficient of damper parallel to the fiber
+    // (normalized to maxIsometricForce)
```

Comment:
b is the symbol in the 2011 paper for width (the non-doxygen) comments right now are the placeholders pointing to that notation.  I am working on the general doxygen muscle description now.  b seems to be more appropriate for damping.  I don't like "w" because it is used in the force-width relationship.  Is there an opensim conventional symbol for width?  I have looked at a couple of muscle and don't see anything.